### PR TITLE
fix: W-19459990 - org picker supports aliases with dashes

### PR DESCRIPTION
**Pull request by CristiCanizales**
### What does this PR do?
This pull request improves how org selections are parsed and handled in the org picker logic, and adds comprehensive tests to ensure correct behavior for various selection formats, including cases with multiple aliases, dashes, and expired indicators.

### Parsing and selection logic improvements

* Enhanced the parsing logic in `OrgList` to accurately extract the username or alias from complex selection strings, including those with multiple dashes, comma-separated aliases, and expired indicators. (`orgList.ts`)

### Test coverage enhancements

* Added a suite of tests for the `setDefaultOrg` method to verify correct command execution for SFDX commands, organization selections with various alias and username formats, and handling of expired org indicators. (`orgList.test.ts`)
* Included tests for edge cases such as selections with no alias, multiple aliases, complex alias structures, and user cancellation. (`orgList.test.ts`)

### What issues does this PR fix or reference?
#6515 , @W-19459990@

### CODE REVIEW

File: packages/salesforcedx-vscode-core/test/jest/orgPicker/orgList.test.ts [hunk 1]
```diff
@@ -357,4 +357,192 @@ describe('OrgList tests', () => {
...
+    });
+
+    it('should handle org with alias containing dashes and expired indicator', async () => {
+      showQuickPickMock.mockResolvedValueOnce('My Organization - Dev Sandbox - foo@bar.com - Expired ❌');
+
+      const result = await orgList.setDefaultOrg();
+
+      expect(executeCommandMock).toHaveBeenCalledWith('sf.config.set', 'My Organization - Dev Sandbox');
+      expect(result).toEqual({ type: 'CONTINUE', data: {} });
+    });
```
Comment (daphne-sfdc) [inline packages/salesforcedx-vscode-core/test/jest/orgPicker/orgList.test.ts:499]: This is the same test case as the previous one on Line 492.

**Review by daphne-sfdc — APPROVED** [review]
- Authorize an org ✅ 
- Set a default org ✅ 
- Create a scratch org (aliases with dashes not allowed) ✅ 
  Screenshot 2025-08-29 at 10 56 01 PM
- Multiple aliases separated by commas ✅ 
- Expired org ✅
